### PR TITLE
pressing enter when cursor is at the end of the line in caption does …

### PR DESCRIPTION
**Pull request by OneByteTwoByte**
…not make an error

### CODE REVIEW

File: src/handlers.ts [hunk 1]
```diff
@@ -129,15 +129,16 @@ function onEnter(editor: Editor, ev: KeyboardEvent): void {
...
     if (caption != null) {
         Editor.withoutNormalizing(editor, () => {
             if (!Range.isCollapsed(selection)) {
                 Editor.deleteFragment(editor)
             }
 
-            Transforms.splitNodes(editor)
-            Transforms.liftNodes(editor, { at: Path.next(captionPath!) })
+            if (!ev.shiftKey) {
+                Editor.insertText(editor, '\n')
```
Comment (aiwenar) [inline src/handlers.ts:140]: A new line (`'\n'`) in this context will be normalized to a space, so this change effectively makes enter and space equivalent.

What the original code intended to do, when an enter is pressed inside a caption, is split the caption in two and move the part after the cursor out of the figure. The important part is moving out of figure, not the visual appearance of a break.

**Review by aiwenar — CHANGES_REQUESTED** [review]

File: src/handlers.ts [hunk 2]
```diff
@@ -135,8 +135,7 @@ function onEnter(editor: Editor, ev: KeyboardEvent): void {
             if (!Range.isCollapsed(selection)) {
                 Editor.deleteFragment(editor)
             }
-
```
Comment (aiwenar) [inline src/handlers.ts:138]: This line was unnecessarily removed.

**Review by aiwenar — APPROVED** [review]
Looks good now.

In future please only force-push once a PR has been approved.

**Comment by aiwenar** [comment]
It might also be good to add a test case that would crash without this fix.

**Comment by OneByteTwoByte** [comment]
I have added test for this issue. It crashes without this fix. Is this ok?

**Review by aiwenar — APPROVED** [review]
> I have added test for this issue. It crashes without this fix. Is this ok?

That was the issue, wasn't it :wink: 

Aside from that, please rebase this PR such that both the fix and test are in a single commit, and it will be good to merge.

**Review by jarosik10 — APPROVED** [review]

**Pull request merged**